### PR TITLE
feat: compareExperiments resolver

### LINKED ISSUE #3475: [experiments] experiment comparison graphql endpoint
```graphql
query compare($datasetId: GlobalID, $expermient_ids: [GlobalID!]) {
  dataset: node(id: $datasetId) {
    compareExperiments(experiments: $experiment_ids) {
      example: Example!
      experimentRuns: [ExperimentRuns]
      warnings: ...
        }
   }
}
```

```
query compare($datasetId: GlobalID!, $experimentIDs: [GlobalID!]!) {
  dataset: node(id: $datasetId) {
    ... on Dataset {
      compareExperiments(experimentIds: $experimentIDs) {
        edges {
          node {
            example
            runs
            containsMultipleRevisions
          }
        }
      }
    }
  }
}
```

`runs` type is a list of union types of either an experiment run or repeated experiment runs.

```
class Run:


```

### PR DESCRIPTION
**Pull request by axiomofjoy**
```graphql
query ($baselineExperimentId: GlobalID!, $comparisonExperimentIds: [GlobalID!]!) {
  compareExperiments(
    baselineExperimentId: $baselineExperimentId
    comparisonExperimentIds: $comparisonExperimentIds
  ) {
    example {
      id
    }
    runComparisonItems {
      experimentId
      runs {
        id
        output
      }
    }
  }
}
```

resolves #3475

### CODE REVIEW

**Comment by axiomofjoy** [comment]
Closing for now.

File: src/phoenix/server/api/types/Dataset.py [hunk 1]
```diff
@@ -236,6 +252,115 @@ async def experiments(
...
+        self,
+        info: Info[Context, None],
+        baseline_experiment_id: GlobalID,
+        comparison_experiment_ids: List[GlobalID],
+    ) -> List[ExperimentComparison]:
+        OrmExample = models.DatasetExample
+        OrmExperiment = models.Experiment
+        OrmRun: TypeAlias = models.ExperimentRun
+        OrmRevision = models.DatasetExampleRevision
+        OrmTrace = models.Trace
```
Comment (mikeldking) [inline src/phoenix/server/api/types/Dataset.py:266]: hoist outside of function scope?

**Comment by mikeldking** [comment]
```graphql
query ($datasetId: GlobalID!, $baselineExperimentId: GlobalID!, $comparisonExperimentIds: [GlobalID!]!) {
        dataset: node(id: $datasetId) {
          ... on Dataset {
            compareExperiments(
              baselineExperimentId: $baselineExperimentId
              comparisonExperimentIds: $comparisonExperimentIds
            ) {
              example {
                id
              }
              runItems {
                experimentId
                runs {
                  id
                  output
                }
              }
            }
          }
        }
      }
```

File: src/phoenix/server/api/types/Dataset.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/server/api/types/Dataset.py]: addressed in 8a5e2450bc330620190667acdb0801f48b5a3479

**Review by mikeldking — APPROVED** [review]

**Pull request merged**